packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageViewWrapper.kt

```diff
@@ -468,6 +482,71 @@ class ExpoImageViewWrapper(context: Context, appContext: AppContext) : ExpoView(
       }
       newTarget.hasSource = sourceToLoad != null
 
+      val downsampleStrategy = if (enableDownscaling) {
+        object : DownsampleStrategy() {
+          var wasTrigger = false
+          override fun getScaleFactor(
+            sourceWidth: Int,
+            sourceHeight: Int,
+            requestedWidth: Int,
+            requestedHeight: Int
+          ): Float {
+            // The method is invoked twice per asset, but we only need to preserve the original dimensions for the first call.
+            // As Glide uses Android downsampling, it can only adjust dimensions by a factor of two,
+            // and hence two distinct scaling factors are computed to achieve greater accuracy.
+            if (!wasTrigger) {
+              newTarget.sourceWidth = sourceWidth
+              newTarget.sourceHeight = sourceHeight
+              wasTrigger = true
+            }
+
+            // The size of the container is unknown, we don't know what to do, so we just run the default scale.
+            if (requestedWidth == SIZE_ORIGINAL || requestedHeight == SIZE_ORIGINAL) {
+              return 1f
+            }
+
+            val rW = requestedWidth.toFloat()
+            val rH = requestedHeight.toFloat()
+            val sW = sourceWidth.toFloat()
+            val sH = sourceHeight.toFloat()
```

Comment:
Using such short names for variables is a bit odd 😅 Can't we just use these `.toFloat()` in the `when` block?